src/crimson/osd/pg_backend.cc

```diff
@@ -473,22 +491,21 @@ seastar::future<> PGBackend::getxattr(
     name = "_" + aname;
   }
   logger().debug("getxattr on obj={} for attr={}", os.oi.soid, name);
-  return getxattr(os.oi.soid, name).then([&osd_op] (ceph::bufferptr val) {
+  return getxattr(os.oi.soid, name).safe_then([&osd_op] (ceph::bufferptr val) {
     osd_op.outdata.clear();
     osd_op.outdata.push_back(std::move(val));
     osd_op.op.xattr.value_len = osd_op.outdata.length();
+    return seastar::now();
     //ctx->delta_stats.num_rd_kb += shift_round_up(osd_op.outdata.length(), 10);
-  }).handle_exception_type(
-    [] (ceph::os::FuturizedStore::EnoentException&) {
-      return seastar::make_exception_future<>(ceph::osd::object_not_found{});
-  }).handle_exception_type(
-    [] (ceph::os::FuturizedStore::EnodataException&) {
-      return seastar::make_exception_future<>(ceph::osd::no_message_available{});
-  });
+  }, ceph::ct_error::enoent::handle([] {
```

Comment:
Ultimately, the goal here would be to modify PGBackend::getxattr to instead return the errorator returned from CyanStore::getxattr so that the next layer up has the same machinery, right?

Reply:
Yes, the handling here is just a makeshift solution. In next phase it will be moved at least to `OpsExecuter::do_osd_op()`.

Reply:
Cool.